agent/internal/detect.go

```diff
@@ -71,11 +73,69 @@ func detectCPUs() ([]device.Device, error) {
 	}
 }
 
+var detectMIGEnabled = []string{
+							"nvidia-smi", "--query-gpu=mig.mode.current", "--format=csv,noheader"}
+var detectNvidiaDevices = []string{"nvidia-smi", "-L"} // Lists both GPUs and MIG instances
```

Comment:
this is a bit unfortunate. i saw `nvidia-smi mig` has some flags for listing but i didn't see anything that supports `--format=csv`, and i can't seem to find any evidence that you can get it from `--query-gpu`..

Reply:
Yeah, I found a lot of inconsistencies when I was trying to see alternative ways of doing this in the CLI. The --id option doesn't get parsed consistently either: --id=[value] is used in one of our commands, but others can't read it unless there's a space intsead of an equals.